ReservedMemory Configuration for NUMA

rthakur-est opened the pull request:
Kubelet throws error when multiple numa nodes are specified for memory
reservation. This is due to "," being used as separator for different
memory types within same numa node as well as for different numa nodes.

This PR fixes the error by using ";" as the separator for
specifying multiple numa node configuration.

Signed-off-by: Ravindra Thakur <ravindra.nath.thakur@est.tech>

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes  #105783

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

rthakur-est commented:
/kind bug

rthakur-est commented:
/sig node

cynepco3hahue commented on pkg/util/flag/flags_test.go:
```diff
@@ -207,7 +207,7 @@ func TestReservedMemoryVar(t *testing.T) {
 		},
 		{
 			desc: "valid input with multiple reserved-memory arguments",
-			argc: "blah --reserved-memory=0:memory=1Gi,hugepages-1Gi=1Gi --reserved-memory=1:memory=1Gi",
```
Can we have both tests?
One for multiple arguments and the second for argument with multiple reservations divided via `;`

rthakur-est committed:
ReservedMemory Configuration for NUMA

Kubelet throws error when multiple numa nodes are specified for memory
reservation. This is due to "," being used as separator for different
memory types within same numa node as well as for different numa nodes.

This PR fixes the error by using ";" as the separator for
specifying multiple numa node configuration.

Signed-off-by: Ravindra Thakur <ravindra.nath.thakur@est.tech>

cynepco3hahue commented:
/ok-to-test

cynepco3hahue commented:
/release-note-none

cynepco3hahue commented:
/priority important-longterm
/triage accepted

cynepco3hahue commented:
/lgtm

dims commented:
/assign @dchen1107 @lavalamp

dims commented:
/approve

swatisehgal commented:
/lgtm

We need to update the documentation to capture this change.

rthakur-est commented:
/retest
